src/shared_modules/indexer_connector/src/indexerConnectorSyncImpl.hpp

```diff
@@ -664,6 +664,52 @@ class IndexerConnectorSyncImpl final
         return resultJson;
     }
 
+    void executeSearchQueryWithPagination(const std::string& index,
+                                         const nlohmann::json& query,
+                                         std::function<void(const nlohmann::json&)> onResponse = nullptr)
+    {
+        nlohmann::json currentQuery = query;
+        std::string searchAfter;
+        while (true)
+        {
+            nlohmann::json searchResult = executeSearchQuery(index, currentQuery);
+
+            // Always call the callback, even for empty pages, to notify the caller of each page's result (including empty pages).
+            if (onResponse)
+            {
+                onResponse(searchResult);
+            }
+
+            // If first page and empty, still break after callback
+            if (!searchResult.contains("hits") || !searchResult["hits"].contains("hits") ||
+                searchResult["hits"]["hits"].empty())
```

Comment:
`searchResult["hits"]["hits"]` is used repeatedly. `operator[]` performs runtime checks and may implicitly insert default values. Extract the array once and validate it. This eliminates accidental mutations and repeated lookups 🦾